### PR TITLE
socket now tries to connect all the time

### PR DESCRIPTION
**Pull request by harelon**
Signed-off-by: Harel <harel.games@gmail.com>

### CODE REVIEW

File: rcv/nes56_cv_manager.py [hunk 1]
```diff
@@ -23,6 +24,7 @@ def __init__(self, conf):
         self._roborio_ip = self._conf.get('rcv_server', 'roborio_ip')
         self._roborio_port = int(self._conf.get('rcv_server', 'roborio_port'))
         self._socket_timeout = int(self._conf.get('rcv_server', 'socket_timeout'))
+        self._connection_status = 'not connected'
```
Comment (bazulay) [inline rcv/nes56_cv_manager.py]: this is a wrong approach"
1 - using a string variable to indicate a state (better use a boolean)
2 - this variable does not own the state .... meaning if you forget to change it (a bug) somewhere the state will not be reflected correctly.

Please read https://stackoverflow.com/questions/35861484/how-to-know-the-if-the-socket-connection-is-closed-in-python especially answer 3 (use fileno on the socket to check whether its connected).
And than all you need to do when sending the data is to check it and if not connected try to reconnect (once) before sending the data.

File: rcv/nes56_cv_manager.py [hunk 2]
```diff
@@ -83,8 +85,10 @@ def connect_to_roborio(self):
                 self._client_socket = None
```
Comment (bazulay) [inline rcv/nes56_cv_manager.py]: It is not a must, but for clearness and consistency you should set here the self._connected to false

File: rcv/rcv_server.py [hunk 3]
```diff
@@ -14,6 +14,7 @@ def exit_gracefully(signum, frame):
 
 
 signal.signal(signal.SIGTERM, exit_gracefully)
+signal.signal(signal.SIGINT, exit_gracefully)
```
Comment (bazulay) [inline rcv/rcv_server.py]: why did you add SIGINT ?

Comment (harelon) [reply rcv/rcv_server.py]: because when i used ctrl c it didn't stop my program and i wanted to close it

**Comment by bazulay** [comment]
Harel,
This patch looks good to me.
However, you didn't write what test you did.
remember to test:
1 -  starting rcv_server when the java side is down (see that it actually starts and tries to connect ... but it shouldn't fail)
2 - starting the java side (and checking that it actually connects and sends data)
3 - stop the java - check that the rcv_server continues to try and connect but dos not fail)
4 - start the java again - check that the rcv_server reconnects and works properly

once this is done you can merge it

**Comment by harelon** [comment]
If i check with a python server is that ok?

**Comment by harelon** [comment]
checked it and it works
